plugins/inputs/bigbluebutton/bigbluebutton_test.go

```diff
@@ -0,0 +1,125 @@
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+var emptyState = false
+
+func getXMLResponse(requestURI string) ([]byte, int) {
+	apiName := strings.Split(strings.TrimPrefix(requestURI, "/bigbluebutton/api/"), "?")[0]
+	xmlFile := fmt.Sprintf("./testdata/%s.xml", apiName)
+
+	if emptyState {
+		xmlFile = fmt.Sprintf("%s.empty_state", xmlFile)
+	}
+
+	code := 200
+	_, err := os.Stat(xmlFile)
+	if err != nil {
+		return nil, 404
+	}
+
+	b, _ := ioutil.ReadFile(xmlFile)
+	return b, code
+}
+
+// return mocked HTTP server
+func getHTTPServer() *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		body, code := getXMLResponse(r.RequestURI)
+		w.WriteHeader(code)
+		if code == 200 {
+			w.Header().Set("Content-Type", "text/xml")
+			w.Write(body)
+		} else {
+			w.Write([]byte(""))
+		}
+	}))
+}
+
+func getPlugin(url string) BigBlueButton {
+	return BigBlueButton{
+		URL:         url,
+		APIEndpoint: "/bigbluebutton/api/",
+		SecretKey:   "OxShRR1sT8FrJZq",
+	}
+}
+
+func TestBigBlueButton(t *testing.T) {
+	s := getHTTPServer()
+	defer s.Close()
+
+	plugin := getPlugin(s.URL)
+	acc := &testutil.Accumulator{}
+	plugin.Gather(acc)
+
+	require.Empty(t, acc.Errors)
+
+	meetingsRecord := map[string]interface{}{
+		"participant_count":       15,
+		"listener_count":          12,
+		"voice_participant_count": 4,
+		"video_count":             1,
+		"active_recording":        0,
+	}
+
+	recordingsRecord := map[string]interface{}{
+		"recordings_count":           2,
+		"published_recordings_count": 1,
+	}
+
+	tags := map[string]string{
+		"server_name": s.URL,
+	}
+
+	require.Equal(t, true, acc.HasMeasurement("bigbluebutton_meetings"))
+	acc.AssertContainsTaggedFields(t, "bigbluebutton_meetings", meetingsRecord, tags)
+
+	require.Equal(t, true, acc.HasMeasurement("bigbluebutton_recordings"))
+	acc.AssertContainsTaggedFields(t, "bigbluebutton_recordings", recordingsRecord, tags)
+}
+
+func TestBigBlueButtonEmptyState(t *testing.T) {
+	emptyState = true
+	s := getHTTPServer()
+	defer s.Close()
+
+	plugin := getPlugin(s.URL)
+	acc := &testutil.Accumulator{}
+	plugin.Gather(acc)
+
+	require.Empty(t, acc.Errors)
+
+	meetingsRecord := map[string]interface{}{
+		"participant_count":       0,
+		"listener_count":          0,
+		"voice_participant_count": 0,
+		"video_count":             0,
+		"active_recording":        0,
+	}
+
+	recordingsRecord := map[string]interface{}{
+		"recordings_count":           0,
+		"published_recordings_count": 0,
+	}
+
+	tags := map[string]string{
+		"server_name": s.URL,
+	}
+
+	require.Equal(t, true, acc.HasMeasurement("bigbluebutton_meetings"))
+	acc.AssertContainsTaggedFields(t, "bigbluebutton_meetings", meetingsRecord, tags)
+
+	require.Equal(t, true, acc.HasMeasurement("bigbluebutton_recordings"))
+	acc.AssertContainsTaggedFields(t, "bigbluebutton_recordings", recordingsRecord, tags)
```

Comment:
Please use `testutil.RequireMetricsEqual()` as this function also outputs a diff which eases the debugging in case of failing tests.

Reply:
Hi @srebhan 

I'm facing an issue with the `testutil.RequireMetricsEqual()`.
My test is :
```go
m, ok := acc.Get("bigbluebutton_meetings")
require.True(t, ok)
expected := testutil.MustMetric("bigbluebutton_meetings", tags, toStringMapInterface(meetingsRecord), time.Unix(0, 0))
testutil.RequireMetricEqual(t, expected, testutil.FromTestMetric(m))
```

Using `time.Unix(0,0)` throws my test with a diff on the time value, and a diff on time format : 
```
-       Time:   s"1970-01-01 00:00:00 +0000 UTC",
+       Time:   s"2021-02-16 08:40:03.585783103 +0000 UTC m=+0.004172704",
```

If I want the same format, I should use `time.Now()` but with the precision, it always throws.

If I'm writing my test like the following, it passes but it's not the best way to do :
```go
m, ok := acc.Get("bigbluebutton_meetings")
require.True(t, ok)
expected := testutil.MustMetric("bigbluebutton_meetings", tags, toStringMapInterface(meetingsRecord), m.Time) // <== Getting time
testutil.RequireMetricEqual(t, expected, testutil.FromTestMetric(m))
```

How can i fix it ?

Thanks,

SLedunois

Reply:
Please specify [`IgnoreTime()`](https://github.com/influxdata/telegraf/blob/a790529bdd9c0f097ce6e7889d8b01bdeb85a4a2/testutil/metric.go#L127) as an option to `RequireMetricEqual`. This will ignore the time when comparing the metrics.